pkg/crosscluster/producer/replication_manager.go

```diff
@@ -475,14 +475,28 @@ func (r *replicationStreamManagerImpl) AuthorizeViaReplicationPriv(
 		}
 
 		// Fallback to legacy REPLICATION priv.
-		if fallbackErr := r.evalCtx.SessionAccessor.CheckPrivilege(ctx,
+		fallbackErr := r.evalCtx.SessionAccessor.CheckPrivilege(ctx,
 			syntheticprivilege.GlobalPrivilegeObject,
-			privilege.REPLICATION); fallbackErr != nil {
-			// We want to return the original error which relates to authorizing with
-			// the REPLICATIONSOURCE priv instead of the error from authorizing with
-			// the deprecated REPLICATION priv.
+			privilege.REPLICATION)
+
+		if fallbackErr == nil {
+			return nil
+		} else if pgerror.GetPGCode(fallbackErr) != pgcode.InsufficientPrivilege {
 			return err
 		}
+
+		// validate VIEWSYSTEMTABLE privilege. This check is used
+		// in debug.zip generation flow as we are going to rely on
+		// restricted privilege access.
+		if fallbackErr = r.evalCtx.SessionAccessor.CheckPrivilege(ctx,
```

Comment:
i don't think this is the right place for this check. This function `AuthorizeViaReplicationPriv` is used to check the replication privileges for all crdb_internal queries related to cross cluster replication. This diff would allow someone with VIEWSTYSTEMTABLE to spin up a replication stream, which would be an escalation of privileges. 

I think you instead need to add this check to the crdb_internal definitions in crdb_internal.go.

Furthermore, are you saying that for these crdb_internal queries must _always_ require the VIEWSYSTEMTABLE priv, in addition to REPLICATIONSOURCE?

Reply:
> Furthermore, are you saying that for these crdb_internal queries must always require the VIEWSYSTEMTABLE priv, in addition to REPLICATIONSOURCE?

We should make sure that the check is "OR" -- we shouldn't require both.